commands/configcmd.go

```diff
@@ -149,22 +148,13 @@ func SetCommand(parent string) *cobra.Command {
 	kustomizeCmd.Example = cfgdocs.SetExamples
 	kustomizeCmd.SilenceUsage = true
 	kustomizeCmd.SilenceErrors = true
-	var autoRun bool
-	setCmd.Flags().BoolVar(&autoRun, "auto-run", true,
-		`Automatically run functions after setting (if enabled for the package)`)
```

Comment:
Instead of commenting on this parameter, let me uplevel and discuss the spirit of this featur. If resources in a package are parameterized, and the user sets those parameters, then the package needs to be re-hydrated i.e. "reconciled". The hydration step is mandatory after _any_ change to resources, not just setters e.g. if you vim edit a file, you need to re-run hydration. Depending on setter and hydration interaction model, we may need to do something similar in `next branch`, where:

kpt cfg set foo bar

is porcelain for:

1. Sett parameter foo to value bar
2. Rerun the hydration pipeline